src/app/auth.guard.ts

```diff
@@ -0,0 +1,51 @@
+import {Injectable} from '@angular/core';
+import {CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router} from '@angular/router';
+import {Observable} from 'rxjs';
+import {AuthService} from './auth.service';
+import {UserService} from './user.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+
+  authToken: any;
+
+  constructor(private auth: AuthService,
+              private router: Router,
+              private user: UserService) {
+  }
+
+  isUserTokenValid(): Promise<boolean> {
+    return this.user.isLoggedIn().toPromise().then(
+      res => {
+        this.authToken = res.token;
+
+        localStorage.removeItem('token');
+        localStorage.setItem('token', this.authToken);
+
+        console.log(this.authToken);
+        return true;
+      },
+      error => {
+        console.log(error.error.message);
```

Comment:
I suppose these will be changed to modals, or toast messages in FE soon? If not please remove console.log